src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/ApiDesignGuidelines/CollectionPropertiesShouldBeReadOnly.cs

```diff
@@ -97,29 +92,37 @@ public override void Initialize(AnalysisContext analysisContext)
             }
 
             // make sure return type is NOT array
-            if (Inherits(property.Type, arrayType))
+            if (Inherits(property.Type, knownTypes.ArrayType))
             {
                 return;
             }
 
             // make sure property type implements ICollection or ICollection<T>
-            if (!Inherits(property.Type, iCollectionType) && !Inherits(property.Type, genericICollectionType))
+            if (!Inherits(property.Type, knownTypes.ICollectionType) && !Inherits(property.Type, knownTypes.GenericICollectionType))
             {
                 return;
             }
 
             // exclude Immutable collections
             // see https://github.com/dotnet/roslyn-analyzers/issues/1900 for details
-            if (!immutableInterfaces.IsEmpty &&
-                property.Type.AllInterfaces.Any(i => immutableInterfaces.Contains(i.OriginalDefinition)))
+            if (!knownTypes.ImmutableInterfaces.IsEmpty &&
+                property.Type.AllInterfaces.Any(i => knownTypes.ImmutableInterfaces.Contains(i.OriginalDefinition)))
+            {
+                return;
+            }
+
+            // exclude readonly collections
+            if (property.Type.OriginalDefinition.Equals(knownTypes.ReadonlyCollection) ||
+                property.Type.OriginalDefinition.Equals(knownTypes.ReadonlyDictionary) ||
+                property.Type.OriginalDefinition.Equals(knownTypes.ReadonlyObservableCollection))
```

Comment:
I have used `Equals` and not `Inherits` because nothing guarantees the behavior for child types.